### PR TITLE
Site Health: adds `WP_DEVELOPMENT_MODE` to site health debug data

### PR DESCRIPTION
**Pull request by kebbet**
https://core.trac.wordpress.org/ticket/58646

### CODE REVIEW

**Comment by spacedmonkey** [comment]
@felixarntz

File: src/wp-admin/includes/class-wp-debug-data.php [hunk 1]
```diff
@@ -243,6 +243,13 @@ public static function debug_data() {
 			$wp_environment_type = __( 'Undefined' );
 		}
 
+		// Check WP_DEVELOPMENT_MODE.
+		if ( defined( 'WP_DEVELOPMENT_MODE' ) && WP_DEVELOPMENT_MODE ) {
+			$wp_development_type = WP_DEVELOPMENT_MODE;
+		} else {
+			$wp_development_type = __( 'Undefined' );
+		}
```
Comment (felixarntz) [inline src/wp-admin/includes/class-wp-debug-data.php]: Not sure the `if` / `else` is needed? The constant is always defined, as it's set in `wp_initial_constants()`.

File: src/wp-admin/includes/class-wp-debug-data.php [hunk 2]
```diff
@@ -323,6 +330,11 @@ public static function debug_data() {
 					'value' => $wp_environment_type,
 					'debug' => $wp_environment_type,
 				),
+				'WP_DEVELOPMENT_MODE' => array(
+					'label' => 'WP_DEVELOPMENT_MODE',
+					'value' => $wp_development_type,
+					'debug' => $wp_development_type,
```
Comment (felixarntz) [inline src/wp-admin/includes/class-wp-debug-data.php]:
```suggestion
					'value' => WP_DEVELOPMENT_MODE ? WP_DEVELOPMENT_MODE : __( 'Disabled' ),
					'debug' => WP_DEVELOPMENT_MODE,
```

**Review by felixarntz — CHANGES_REQUESTED** [review]
@kebbet Two small points of feedback here: No need to check if the constant is defined as it's always defined, similar to e.g. `WP_DEBUG`.

**Comment by kebbet** [comment]
Changes applied and trunk merged into branch.

**Review by felixarntz — APPROVED** [review]
Thanks @kebbet, LGTM!

**Comment by felixarntz** [comment]
Committed in https://core.trac.wordpress.org/changeset/56079

**Pull request closed without merging**